src/webservice/Router.cpp

```diff
@@ -0,0 +1,125 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "webservice/Router.h"
+
+#include <sstream>
+
+#include <proxygen/httpserver/RequestHandler.h>
+
+#include "webservice/NotFoundHandler.h"
+
+namespace nebula {
+
+namespace web {
+
+void Route::checkPath(const std::string &path) {
+    CHECK(!path.empty() && path[0] == '/') << "Path must start with '/'";
+}
+
+void Route::setPath(const std::string &path) {
+    auto loc = path.find(':');
+    if (loc != std::string::npos) {
+        createPattenRegex(path);
+        this->path_ = path;
+    } else {
+        // exact path
+        if (path.back() == '/') {
+            this->path_ = path.substr(0, path.length() - 1);
+        } else {
+            this->path_ = path;
+        }
+    }
+}
+
+std::unique_ptr<std::regex> Route::reToken_ =
```

Comment:
add `const`

Reply:
this is static variables

Reply:
@laura-ding  @dangleptr  I have fixed all suggest issues. please review again!